src/eval.rs

```diff
@@ -199,6 +203,7 @@ pub fn eval_main<'tcx>(tcx: TyCtxt<'tcx>, main_id: DefId, config: MiriConfig) {
             if !ignore_leaks && leaks != 0 {
                 tcx.sess.err("the evaluated program leaked memory");
             }
+            std::process::exit(return_code as i32);
```

Comment:
Instead of calling `exit` here (and in the `InterpError::Exit` case below), IMO it would make more sense to let `eval_main` return the return code and have the caller exit.

Also, this `as` cast here could fail. At the least, please use `i32::try_from`.

Reply:
This also broke the leak check, leading to some tests failing.